RealmSwift/Tests/ModernObjectTests.swift

```diff
@@ -134,6 +134,9 @@ class ModernObjectTests: TestCase {
         test(ModernPrimaryObjectIdObject(), ObjectId.generate(), ObjectId.generate())
         test(ModernPrimaryOptionalObjectIdObject(), ObjectId.generate(), nil)
 
+        test(ModernPrimaryIntEnumObject(), .value1, .value2)
+        test(ModernPrimaryOptionalIntEnumObject(), .value1, nil)
```

Comment:
Would it be worth adding another modern test object which declares all indexable types and then iterates over the schema to make sure they are indeed indexed?

Reply:
I've added a test which verifies that all indexable types are marked as indexed in the schema when they should be.